packages/flutter/lib/src/rendering/semantics_event.dart

```diff
@@ -121,3 +121,31 @@ class ScrollCompletedSemanticsEvent extends SemanticsEvent {
     return map;
   }
 }
+
+/// An event for a semantic announcement.
+///
+/// This should be used for announcement that are not seamlessly announced by
+/// the system as a result of a UI state change.
+///
+/// For example a camera application can use this method to make accessibility
+/// announcements regarding objects in the viewfinder.
+class AnnounceSemanticsEvent extends SemanticsEvent {
+
+  AnnounceSemanticsEvent(this.message, this.textDirection) : super('announce');
```

Comment:
docs

Reply:
Not sure what to document here...
The parameters are documented on the members, do we want to re-describe them here?

Looking at what we document for e.g the constructor for SemanticsEvent we say ("Initializes internal fields."), do you want to say something like that? (I kind of feel this is obvious...)
And then we say something about the parameter (which is then re-described on the member).